### DISCUSSION TITLE
Usage issues：use_enum_values

### DISCUSSION THREAD

**Discussion by 212413616** [pydantic/pydantic#11110] [Question]
Hi,
I have encountered a problem

```
import enum

from pydantic import BaseModel


class Sex(enum.StrEnum):
    Male = "male"
    Female = "female"


class User(BaseModel):
    name: str
    age: int
    sex: Sex

    class Config:
        use_enum_values = True


user = User(name="Alice", age=18, sex=Sex.Male)

print(user.model_dump())  # output: {"name": "Alice", "age": 18, "sex": "male"}

print(type(user.sex))  # output: <class 'str'>
print(user.sex.value)  # output: AttributeError: 'str' object has no attribute 'value'


```

My English is not very good, please forgive me for asking in Chinese

- English
﻿
Adding `use_denum_falues=True` and using `user. model_fump()` can meet my needs
﻿
However, I still need to output `<enum'Sex'>` when using `type (user. sex)` because I need to use `user. sex. value `. 
﻿
How should I write the code

Thank you
﻿
﻿

- 中文
加上`use_enum_values = True`，使用`user.model_dump()`能够满足我的需求

但是，我还需要在`type(user.sex)`时输出`<enum 'Sex'>`，因为我需要使用`user.sex.value`，请问应该怎么编写代码，谢谢

**Comment by janas-adam** [comment]
hi,
don't know if i understood ur question well:
1. u have to leave `use_enum_values = True`
2. u want to have enum-typed return 

so maybe extend it like:
```
class User(BaseModel):
    name: str
    age: int
    sex: Sex

    class Config:
        use_enum_values = True

    def get_enum_value(self):
        return Sex(self.sex)
```

which ll give u
`>>type(user.get_enum_value()) # output <enum 'Sex'>`

**Reply by 212413616** [reply]
> hi, 你好， don't know if i understood ur question well:不知道我是否很好地理解了你的问题：
> 
> 1. u have to leave `use_enum_values = True`你必须离开 `use_enum_values = True`
> 2. u want to have enum-typed return你想要枚举类型的返回值
> 
> so maybe extend it like:所以也许可以将其扩展为：
> 
> ```
> class User(BaseModel):
>     name: str
>     age: int
>     sex: Sex
> 
>     class Config:
>         use_enum_values = True
> 
>     def get_enum_value(self):
>         return Sex(self.sex)
> ```
> 
> which ll give u 这会给你 `>>type(user.get_enum_value()) # output <enum 'Sex'>`





Thank you very much for your answer

If class ```User``` had many enum attributes

The code written will be very cumbersome

Like below
```
import enum

from pydantic import BaseModel


class Sex(enum.StrEnum):
    Male = "male"
    Female = "female"


class OtherEnum(enum.StrEnum):
    xxx = "xxx"


class OtherEnum2(enum.StrEnum):
    xxx = "xxx"


class User(BaseModel):
    name: str
    age: int
    sex: Sex

    other: OtherEnum
    other2: OtherEnum2

    class Config:
        use_enum_values = True

    def get_sex_enum_value(self):
        return Sex(self.sex)

    def get_other_enum_value(self):
        return OtherEnum(self.sex)

    def get_other2_enum_value(self):
        return OtherEnum2(self.sex)
```
What I hope for is as follows

```
user = User(name="Alice", age=18, sex=Sex.Male)

print(type(user.sex))  #  hope output: <class Sex>, but real output: <class 'str'>    
print(user.sex.value)  # output: AttributeError: 'str' object has no attribute 'value'
```

Do you have any better ideas,I am looking forward to hearing from you again

(Sorry for my poor English)

**Reply by janas-adam** [reply]
why u can't just delete config with "use_enum_values" from the class ?
